sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -385,6 +385,16 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val SHUFFLEDHASHJOIN_FALLBACK_ENABLED =
+    buildConf("spark.sql.join.enableShuffledHashJoinFallback")
+      .internal()
+      .doc("When true, enable sort-based fallback for shuffled hash join. " +
+        "The sort-based fallback is to use sort merge join when there is no enough memory " +
+        "to build hash table for shuffled hash join. This can help avoid task OOM automatically.")
+      .version("3.2.0")
+      .booleanConf
+      .createWithDefault(false)
```

Comment:
Seems a bit overkill to add a config for this. People can just flip the existing config to prefer SMJ, then SHJ will never be used.